XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,399 @@
+				videoStabilizationEffect = (VideoStabilizationEffect)await mediaCapture.AddVideoEffectAsync(stabilizerDefinition, MediaStreamType.VideoRecord);
+				var recommendation = videoStabilizationEffect.GetRecommendedStreamConfiguration(mediaCapture.VideoDeviceController, encodingProfile.Video);
+				if (recommendation.InputProperties != null)
+				{
+					inputPropertiesBackup = mediaCapture.VideoDeviceController.GetMediaStreamProperties(MediaStreamType.VideoRecord) as VideoEncodingProperties;
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, recommendation.InputProperties);
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoPreview, recommendation.InputProperties);
+				}
+				if (recommendation.OutputProperties != null)
+				{
+					outputPropertiesBackup = encodingProfile.Video;
+					encodingProfile.Video = recommendation.OutputProperties;
+				}
+				videoStabilizationEffect.Enabled = true;
+			}
+			mediaRecording = await mediaCapture.PrepareLowLagRecordToStorageFileAsync(encodingProfile, file);
+			await mediaRecording.StartAsync();
+		}
+
+		async Task<MediaSource> StopRecord()
+		{
+			if (mediaRecording == null)
+				return null;
+
+			await mediaRecording.StopAsync();
+			await mediaRecording.FinishAsync();
+			mediaRecording = null;
+
+			if (videoStabilizationEffect != null)
+			{
+				await mediaCapture.RemoveEffectAsync(videoStabilizationEffect);
+				videoStabilizationEffect = null;
+
+				if (inputPropertiesBackup != null)
+				{
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, inputPropertiesBackup);
+					inputPropertiesBackup = null;
+				}
+				if (outputPropertiesBackup != null)
+				{
+					encodingProfile.Video = outputPropertiesBackup;
+					outputPropertiesBackup = null;
+				}
+				videoStabilizationEffect = null;
+			}
+
+			return MediaSource.FromFile(filePath);
+		}
+
+		void MediaCaptureFailed(MediaCapture sender, MediaCaptureFailedEventArgs errorEventArgs)
+			=> Element?.RaiseMediaCaptureFailed(errorEventArgs.Message);
+
+		protected override async void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				case nameof(CameraView.CameraOptions):
+					await CleanupCameraAsync();
+					await InitializeCameraAsync();
+					break;
+				case nameof(CameraView.FlashMode):
+					if (flash != null)
+						flash.IsEnabled = Element.FlashMode == CameraFlashMode.Torch || Element.FlashMode == CameraFlashMode.On;
+					break;
+				case nameof(CameraView.PreviewAspect):
+					// TODO
+					break;
+				case nameof(CameraView.Zoom):
+					UpdateZoom();
+					break;
+			}
+			base.OnElementPropertyChanged(sender, e);
+		}
+
+		void UpdateZoom()
+		{
+			var zoomControl = mediaCapture.VideoDeviceController.ZoomControl;
+			if (!zoomControl.Supported)
```

Comment:
Just wondering if we should create a BindableProperty to expose this to the dev. To makes sense to have a boolean `IsZoomSupported`. What do you think? @bmacombe @jfversluis

Reply:
There is a MaxZoom property, which returns 1 when no zoom is available.  Could probably add that just in CameraView, true if MaxZoom >1 otherwise false.